### PR TITLE
MODPUBSUB-107: Set/update lastName for the pub-sub user.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MODPUBSUB-107 - Show **system** as source in loan action table when lost item fees are paid

## Purpose

Since mod-circulation integration with pub-sub it is required to have a last name for the user in order to properly display source of action at loan action page.

Emma has discussed it with RA SIG and they decided to use **System** as an human identifier of the user (i.e. the last name).

## Changes

* Add `personal.lastName` property to user json that used [to create the user](https://github.com/folio-org/mod-pubsub/pull/88/files#diff-f11076db34e5ecca90023354b89b08baR250);
* [Do update](https://github.com/folio-org/mod-pubsub/pull/88/files#diff-f11076db34e5ecca90023354b89b08baR142) of the existing user if the property is not present (to cover migration case);
* Add some tests to verify the changes;
* [Forward user update failure](https://github.com/folio-org/mod-pubsub/pull/88/files#diff-7455f115aea6bcad69b2af2de5ec9e11) to `/_/tenant` response and verify it.

### CODE REVIEW

**Comment by bohdan-suprun** [comment]
Hi @OleksiiKuzminov, @KaterynaSenchenko, @RuslanLavrov could you please review the changes, when you get a chance.

File: mod-pubsub-server/src/main/java/org/folio/representation/User.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package org.folio.representation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+
+public class User {
```
Comment (marcjohnson-kint) [inline mod-pubsub-server/src/main/java/org/folio/representation/User.java:9]: I believe lombok has been introduced in other places to reduce some of this boilerplate code. Could it be introduced here as well?

Comment (bohdan-suprun) [reply mod-pubsub-server/src/main/java/org/folio/representation/User.java:9]: I'm fine to introduce lombok here as well. unless the team has any objections.

Comment (bohdan-suprun) [reply mod-pubsub-server/src/main/java/org/folio/representation/User.java:9]: @marcjohnson-kint, I'll leave to the team to add this dependency. Having it just for a single class not give much benefit.

**Pull request merged**